File: be/src/labels/entities/label.entity.ts
import { Project } from 'src/projects/entities/project.entity';
import { Entity, PrimaryGeneratedColumn, Column as ColumnField, OneToMany, ManyToMany, JoinTable, JoinColumn, ManyToOne } from 'typeorm';

@Entity('labels')
export class Label {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @ColumnField({ name: 'name' })
  name: string;

  @ColumnField({ default: '#007bff', name: 'color' })
  color: string;

  @ColumnField({ type: 'text', nullable: true, name: 'description' })
  description?: string;

  @ManyToOne(()=>Project, (project)=>project.labels, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'project_id' })
  project: Project;
}

File: backend-java/src/main/java/com/qlda/backendjava/labels/entity/LabelEntity.java
package com.qlda.backendjava.labels.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.qlda.backendjava.projects.entity.ProjectEntity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

@Entity
@Table(name = "labels")
@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class LabelEntity {

    @Id
    @GeneratedValue(generator = "uuid2")
    @GenericGenerator(name = "uuid2", strategy = "uuid2")
    @Column(columnDefinition = "VARCHAR(36)")
    private String id;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "color", nullable = false)
    private String color = "#007bff";

    @Column(name = "description", columnDefinition = "TEXT", nullable = true)
    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "project_id", nullable = false)
    @JsonIgnore
    private ProjectEntity project;
}

